mongo/integration/initial_dns_seedlist_discovery_test.go

```diff
@@ -230,6 +262,14 @@ func getServerByAddress(address string, topo *topology.Topology) (description.Se
 	if err != nil {
 		return description.Server{}, err
 	}
+
+	// If the selected server is a topology.SelectedServer, then we can get the description without creating a
```

Comment:
This is a really good question! Strangely, "Topology.SelectServer" returns an interface for a driver server, but AFAIK all paths of the method lead to a server that results from "Topology.FindServer" which returns a "SelectedServer" struct. So unless we mock the SelectServer method somewhere, I think we would always get a "topology.SelectedServer".

I'm pretty sure the "ok" check for this block is not necessary, but I went that direction because it is defensive in case we do mock this method at some point in the future.

Reply:
> I'm pretty sure the "ok" check for this block is not necessary, but I went that direction because it is defensive in case we do mock this method at some point in the future.

Thanks for the explanation; I'm fine to leave the check in. [Here](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/unified_runner_events_helper_test.go#L169) is an example of where we do something similar in testing but type assert without an if-check.